deeplearning4j-keras/src/main/java/org/deeplearning4j/keras/DeepLearning4jEntryPoint.java

```diff
@@ -0,0 +1,45 @@
+package org.deeplearning4j.keras;
+
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+/**
+ * API exposed to the Python side. This class contains methods which are used by the python wrapper.
+ * It is instantiated directly in the server code.
+ */
+@Slf4j
+public class DeepLearning4jEntryPoint {
+
+    private final NeuralNetworkReader neuralNetworkReader = new NeuralNetworkReader();
+    private final NDArrayHDF5Reader ndArrayHDF5Reader = new NDArrayHDF5Reader();
+    private final NeuralNetworkBatchLearner neuralNetworkBatchLearner = new NeuralNetworkBatchLearner();
+
+    /**
+     * Performs fitting of the model which is referenced in the parameters according to learning parameters specified.
+     *
+     * @param entryPointFitParameters Definition of the model and learning process
+     */
+    public void fit(EntryPointFitParameters entryPointFitParameters) throws Exception {
+
+        try {
+            MultiLayerNetwork multiLayerNetwork = neuralNetworkReader.readNeuralNetwork(entryPointFitParameters);
+
+            INDArray features = ndArrayHDF5Reader.readFromPath(entryPointFitParameters.getTrainFeaturesFile());
```

Comment:
No - on Python side I'm dumping the dataset myself (and I chose HDF5 - this is the same format Keras uses to export models).

I know that there are problems with this e.g. de/serialization slows down the process, there are obvious practical size limitations. The reason I did it this way is that on the Python side the data coming from the raw data set can be pre-processed (and I think it usually is :) ) before learning and we would have to somehow copy that logic to Java in order to be able to perform the fitting. 

The problem here is that we are not using Jython - because it cannot support numpy, and simply just export/import the Keras model. This requires exporting/importing the pre-processed data which was supposed to be fed into the model. 

For large distributed datasets it won't work completely - however I'm not sure if in this case someone wouldn't just invest time in rewriting his model in DL4J directly.

Regarding reading/writing numpy arrays: the format used here is pickle (https://docs.python.org/2/library/pickle.html). I was looking for some decent library to read this inside JVM and couldn't really find anything what would convince me that it is stable and well tested. On the other hand HDF5 is already used by Keras and seems to work pretty well.

I don't really have a strong opinion which lib/data format should be used - if you feel there is a better way to do it let me know.